File: Store/Store.Web/Modules/Store/OperationType/OperationTypeForm.cs

namespace Store.Store.Forms
{
    using Serenity.ComponentModel;
    using System;

    [FormScript("Store.OperationType")]
    [BasedOnRow(typeof(Entities.OperationTypeRow), CheckNames = true)]
    public class OperationTypeForm
    {
        public Int16 OpCode { get; set; }
        public String Operation { get; set; }
    }
}

File: Ser2023/Store/Store.Web/Modules/ServerTypes/Store/OperationTypeForm.ts
import { IntegerEditor, StringEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface OperationTypeForm {
    OpCode: IntegerEditor;
    Operation: StringEditor;
}

export class OperationTypeForm extends PrefixedContext {
    static formKey = 'Store.OperationType';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!OperationTypeForm.init)  {
            OperationTypeForm.init = true;

            var w0 = IntegerEditor;
            var w1 = StringEditor;

            initFormType(OperationTypeForm, [
                'OpCode', w0,
                'Operation', w1
            ]);
        }
    }
}
